govcd/nsxt_edgegateway.go

```diff
@@ -0,0 +1,301 @@
+	if egw.EdgeGateway.OrgVdc.ID != vdc.Vdc.ID {
+		return nil, fmt.Errorf("%s: no NSX-T edge gateway with ID '%s' found in VDC '%s'",
+			ErrorEntityNotFound, id, vdc.Vdc.ID)
+	}
+
+	return egw, nil
+}
+
+// GetNsxtEdgeGatewayByName allows to retrieve NSX-T edge gateway by Name for Org admins
+func (adminOrg *AdminOrg) GetNsxtEdgeGatewayByName(name string) (*NsxtEdgeGateway, error) {
+	queryParameters := url.Values{}
+	queryParameters.Add("filter", "name=="+name)
+
+	allEdges, err := adminOrg.GetAllNsxtEdgeGateways(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve edge gateway by name '%s': %s", name, err)
+	}
+
+	onlyNsxtEdges := filterOnlyNsxtEdges(allEdges)
+
+	return returnSingleNsxtEdgeGateway(name, onlyNsxtEdges)
+}
+
+// GetNsxtEdgeGatewayByName allows to retrieve NSX-T edge gateway by Name for Org admins
+func (org *Org) GetNsxtEdgeGatewayByName(name string) (*NsxtEdgeGateway, error) {
+	queryParameters := url.Values{}
+	queryParameters.Add("filter", "name=="+name)
+
+	allEdges, err := org.GetAllNsxtEdgeGateways(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve edge gateway by name '%s': %s", name, err)
+	}
+
+	onlyNsxtEdges := filterOnlyNsxtEdges(allEdges)
+
+	return returnSingleNsxtEdgeGateway(name, onlyNsxtEdges)
+}
+
+// GetNsxtEdgeGatewayByName allows to retrieve NSX-T edge gateway by Name for specifi Vdc
+func (vdc *Vdc) GetNsxtEdgeGatewayByName(name string) (*NsxtEdgeGateway, error) {
+	queryParameters := url.Values{}
+	queryParameters.Add("filter", "name=="+name)
+
+	allEdges, err := vdc.GetAllNsxtEdgeGateways(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve edge gateway by name '%s': %s", name, err)
+	}
+
+	onlyNsxtEdges := filterOnlyNsxtEdges(allEdges)
+
+	return returnSingleNsxtEdgeGateway(name, onlyNsxtEdges)
+}
+
+// GetAllNsxtEdgeGateways allows to retrieve all NSX-T edge gateways for Org Admins
+func (adminOrg *AdminOrg) GetAllNsxtEdgeGateways(queryParameters url.Values) ([]*NsxtEdgeGateway, error) {
+	return getAllNsxtEdgeGateways(adminOrg.client, queryParameters)
+}
+
+// GetAllNsxtEdgeGateways  allows to retrieve all NSX-T edge gateways for Org users
+func (org *Org) GetAllNsxtEdgeGateways(queryParameters url.Values) ([]*NsxtEdgeGateway, error) {
+	return getAllNsxtEdgeGateways(org.client, queryParameters)
+}
+
+// GetAllNsxtEdgeGateways  allows to retrieve all NSX-T edge gateways for Org users
+func (vdc *Vdc) GetAllNsxtEdgeGateways(queryParameters url.Values) ([]*NsxtEdgeGateway, error) {
+	filteredQueryParams := queryParameterFilterAnd("orgVdc.id=="+vdc.Vdc.ID, queryParameters)
+	return getAllNsxtEdgeGateways(vdc.client, filteredQueryParams)
+}
+
+// CreateNsxtEdgeGateway allows to create NSX-T edge gateway for Org admins
+func (adminOrg *AdminOrg) CreateNsxtEdgeGateway(e *types.OpenAPIEdgeGateway) (*NsxtEdgeGateway, error) {
+	if !adminOrg.client.IsSysAdmin {
+		return nil, fmt.Errorf("only Provider can create Edge Gateway")
```

Comment:
Not sure this message is correct:) Maybe better "System Administrator" ?

Reply:
Ok.